question-service/src/main/java/com/information/center/questionservice/service/QuestionServiceImpl.java

```diff
@@ -2,122 +2,139 @@
 
 import com.information.center.questionservice.converter.QuestionConverter;
 import com.information.center.questionservice.entity.QuestionEntity;
+import com.information.center.questionservice.model.QuestionListDetails;
 import com.information.center.questionservice.model.request.AnswerRequest;
 import com.information.center.questionservice.model.request.QuestionRequest;
 import com.information.center.questionservice.model.response.AnswerResponse;
 import com.information.center.questionservice.model.response.QuestionResponse;
+import com.information.center.questionservice.model.response.QuestionResponsePage;
 import com.information.center.questionservice.repository.QuestionRepository;
 import exception.MicroserviceException;
-import java.util.List;
-import java.util.UUID;
-import java.util.function.Supplier;
 import lombok.RequiredArgsConstructor;
+import lombok.var;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Date;
+import java.util.UUID;
+import java.util.function.Supplier;
 
 @Component
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class QuestionServiceImpl implements QuestionService {
 
-  private final QuestionRepository questionRepository;
-  private final QuestionConverter questionConverter;
+    private final QuestionRepository questionRepository;
+
+    private final QuestionConverter questionConverter;
+
+    private final AnswerService answerService;
+
+    private final RestTemplate template;
 
-  @Override
-  public QuestionResponse create(QuestionRequest questionRequest, String topicExternalId) {
+    private static final String ORGANISATION_ENDPOINT = "/topic/internal/";
 
-    String questionExternalId = UUID.randomUUID().toString();
+    @Value("${endpoints.topic:http://localhost:8881}")
+    private String equipmentEndpoint;
 
-    for (AnswerRequest answerRequest : questionRequest.getAnswers()) {
-      answerRequest.setQuestionExternalId(questionExternalId);
-      answerRequest.setExternalId(UUID.randomUUID().toString());
+    @Override
+    public String getTopicNameByExternalId(String topicId) {
```

Comment:
if you want to call the topic-service you can use `@FeignClient`. You can find some example in account-service